safehttp/mux.go

```diff
@@ -71,18 +71,48 @@ type ServeMux struct {
 	mux *http.ServeMux
 }
 
-func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]handlerConfig) {
-	for pattern, handlersPerMethod := range handlers {
-		handlersPerMethod := handlersPerMethod
-		mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-			cfg, ok := handlersPerMethod[r.Method]
-			if !ok {
-				http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
-				return
-			}
-			processRequest(cfg, w, r)
-		})
+func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]*handlerConfig) {
+	for pattern, handlerConfigPerMethod := range handlers {
+		httpHandler := stdHandler(handlerConfigPerMethod)
+		mux.Handle(pattern, httpHandler)
+	}
+}
+
+// stdHandler creates an http.Handler using a handlerConfig.
+// The handlerConfig map is needed because http.Handle handles requests for any HTTP request method,
+// which we want to avoid in safehttp.
+func stdHandler(handlerConfigPerMethod map[string]*handlerConfig) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		cfg, ok := handlerConfigPerMethod[r.Method]
+		if !ok {
+			http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+			return
+		}
+		processRequest(cfg, w, r)
+	})
+}
+
+// TODO rename...
+type HandlerRegistration struct {
+	Method  string
+	Handler Handler
+	Cfgs    []InterceptorConfig
+}
+
+// StdHandler creates an http.Handler using the ServeMuxConfig
+// (i.e. using already installed Interceptors etc).
+// This is intended to be used during transition from using the http package to safehttp.
+func (s *ServeMuxConfig) StdHandler(hr []*HandlerRegistration) http.Handler {
```

Comment:
`hrs ...*HandlerRegistration`

Reply:
Of course, the `ServeMuxConfig` should not have the `StdHandler` method and it should belong to a different type instead. But my understanding is that you'll proceed with this once you get an LGTM on the idea. :)

Reply:
That's the plan :)